### PR TITLE
add etag and other tests

### PR DESCRIPTION
**Pull request by mchangrh**

### CODE REVIEW

File: src/routes/getSegmentInfo.ts [hunk 1]
```diff
@@ -34,11 +34,11 @@ async function handleGetSegmentInfo(req: Request, res: Response): Promise<DBSegm
     // deduplicate with set
     UUIDs = [ ...new Set(UUIDs)];
     // if more than 10 entries, slice
-    if (UUIDs.length > 10) UUIDs = UUIDs.slice(0, 10);
-    if (!Array.isArray(UUIDs) || !UUIDs) {
+    if (!Array.isArray(UUIDs) || !UUIDs?.length) {
         res.status(400).send("UUIDs parameter does not match format requirements.");
         return;
     }
+    if (UUIDs.length > 10) UUIDs = UUIDs.slice(0, 10);
```
Comment (ajayyy) [inline src/routes/getSegmentInfo.ts:41]: why?

File: test/utils/reset.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { Pool } from "pg";
+import { Logger } from "../../src/utils/logger";
+
+export async function resetRedis() {
+    if (config?.redis?.enabled) {
+        const client = createClient(config.redis);
+        await client.connect();
+        await client.flushAll();
+    }
+}
```
Comment (ajayyy) [inline test/utils/reset.ts:14]: no check if we are testing?

File: src/routes/getSegmentInfo.ts (hunk 1 above)
Comment (mchangrh) [reply src/routes/getSegmentInfo.ts:41]: this way we can throw a proper 400 instead of 500 when TypeError occurs

File: test/utils/reset.ts (hunk 2 above)
Comment (mchangrh) [reply test/utils/reset.ts]: add explicit tests

File: src/routes/getSegmentInfo.ts (hunk 1 above)
Comment (ajayyy) [reply src/routes/getSegmentInfo.ts:41]: I'm confused, isn't this saying that if it is too long, we just ignore the extra stuff?

Comment (mchangrh) [reply src/routes/getSegmentInfo.ts:41]: This was part of the original segmentInfo, to only batch 10 UUIDs at at time, it was just moved to be **after** the isArray check instead

Comment (mchangrh) [reply src/routes/getSegmentInfo.ts:41]: for reference: https://discord.com/channels/603643120093233162/607338052221665320/854440486453510206

**Pull request merged**